hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerRequestHandler.java

```diff
@@ -307,9 +309,27 @@ public OMClientResponse handleWriteRequest(OMRequest omRequest,
     OMClientResponse omClientResponse = null;
     omClientRequest =
         OzoneManagerRatisUtils.createClientRequest(omRequest, impl);
-    omClientResponse = omClientRequest
-        .validateAndUpdateCache(getOzoneManager(), transactionLogIndex,
-            ozoneManagerDoubleBuffer::add);
+    try {
+      omClientResponse = omClientRequest
+          .validateAndUpdateCache(getOzoneManager(), transactionLogIndex,
+              ozoneManagerDoubleBuffer::add);
+    } catch (InvalidPathException e) {
+      OMException omException = new OMException(e.getMessage(),
+          OMException.ResultCodes.INVALID_PATH);
+      OMResponse.Builder omResponse =
+          OmResponseUtil.getOMResponseBuilder(omRequest);
+      omResponse.setSuccess(false);
+      omResponse.setMessage(omException.getMessage());
+      omResponse.setStatus(OzoneManagerRatisUtils.exceptionToResponseStatus(
+          omException));
+      omClientResponse =
+          new InvalidPathClientResponse(omResponse.build());
+      if (omClientResponse.getFlushFuture() == null) {
+        omClientResponse.setFlushFuture(
+            ozoneManagerDoubleBuffer.add(omClientResponse,
+                transactionLogIndex));
```

Comment:
I am not sure why we need to do this

Reply:
There was a inconsistence of the transactionId among the OMs, followers have problem starting without these lines.

Reply:
@symious , thanks for continue working on this patch.  Can you explain it a bit more about why we need to add this omClientResponse.getFlushFuture() check?

Reply:
I think I added it because the flustFuture might be already set in "validateAndUpdateCache()", so if it's already set, we don't need to do it again.

Reply:
When InvalidPathException is handled in handleWriteRequest, instead of validateAndUpdateCache, there will be no audit log at all for this request. Since audit log is important, I would suggest move the InvalidPathException into validateAndUpdateCache if possible.